libnemo-private/nemo-context-menu-menu-item.c

```diff
@@ -0,0 +1,431 @@
+/* GTK - The GIMP Toolkit
+ * Copyright (C) 1995-1997 Peter Mattis, Spencer Kimball and Josh MacDonald
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <config.h>
+
+#include "nemo-global-preferences.h"
+#include "nemo-context-menu-menu-item.h"
+#include "nemo-widget-action.h"
+
+#include <glib/gi18n.h>
+
+static GtkActivatableIface *parent_activatable_iface;
+
+static void nemo_context_menu_menu_item_dispose              (GObject *object);
+
+static gboolean nemo_context_menu_menu_item_enter      (GtkWidget        *widget,
+                                          GdkEventCrossing *event);
+static gboolean nemo_context_menu_menu_item_leave      (GtkWidget        *widget,
+                                          GdkEventCrossing *event);
+static gboolean nemo_context_menu_menu_item_motion     (GtkWidget        *widget,
+                                          GdkEventMotion   *event);
+
+static gboolean nemo_context_menu_menu_item_button_press (GtkWidget      *widget,
+                                                          GdkEventButton *event);
+
+static gboolean nemo_context_menu_menu_item_button_release (GtkWidget      *widget,
+                                                            GdkEventButton *event);
+
+static void nemo_context_menu_menu_item_set_label (GtkMenuItem      *menu_item,
+                                                   const gchar      *label);
+
+static void nemo_context_menu_menu_item_activatable_interface_init (GtkActivatableIface  *iface);
+
+G_DEFINE_TYPE_WITH_CODE (NemoContextMenuMenuItem, nemo_context_menu_menu_item, GTK_TYPE_IMAGE_MENU_ITEM,
+                         G_IMPLEMENT_INTERFACE (GTK_TYPE_ACTIVATABLE,
+                                                nemo_context_menu_menu_item_activatable_interface_init));
+
+static void
+update_toggle_state (NemoContextMenuMenuItem *item, gboolean from_event)
+{
+    gboolean complex_mode = g_settings_get_boolean (nemo_preferences, NEMO_PREFERENCES_CONTEXT_MENUS_SHOW_ALL_ACTIONS);
+
+    gtk_label_set_text (GTK_LABEL (item->toggle_label_widget),
+                        complex_mode ? _("Show less actions") : _("Show more actions"));
```

Comment:
I didn't want to upset translations at this point (I'll probably try to get this in 2.8.x still) - but you're right, it could be worded better I think